### PR TITLE
docs(issue-26): Documentation for action testing

### PR DESCRIPTION
**Pull request by ferborva**
- simple actions
- async actions
- composed actions
- note of caution

### CODE REVIEW

File: docs/testing/actions.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
+## Simple actions
+
+You will often find yourself writting actions that hold no special internal logic. These will likely only call `setState` or do so conditionally based on inputs. For example:
+
+```js
+const actions = {
+  reset: () => ({ setState, getState }, { canReset }) => {
+      if (canReset && getState().count !== 0) {
```
[the remark this answers is not in the document]
Comment (ferborva) [reply docs/testing/actions.md]: Why would anyone call the action with `false`? Hypothetically:

```
// I can image this
if (shouldReset) {
    reset();
}

// Harder to imagine this
reset(shouldReset);
```
🤷‍♂

[the remark this answers is not in the document]
Comment (albertogasparin) [reply docs/testing/actions.md]: Fair enough, can we find a better example then? 🙏

File: docs/testing/actions.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+      return newUserData;
+    } catch (e) {
+      // Store failure and throw
+      setState({
+        loading: false,
+        error: e.message,
+      });
+
+      throw new Error(e.message);
```
[the remark this answers is not in the document]
Comment (ferborva) [reply docs/testing/actions.md]: Given this is an async action returning data, my thought was allowing the consumer to:
```
actionCall().then().catch();
```
If it doesn't throw then the return should always send back the `{ data, error }`. Which I guess also makes sense.
```
actionCall().then(({data, error}) => { if (error) ... else ... })
```

[the remark this answers is not in the document]
Comment (albertogasparin) [reply docs/testing/actions.md]: Yes but that is an anti-pattern with hooks and render props :)

[the remark this answers is not in the document]
Comment (ferborva) [reply docs/testing/actions.md]: Very true. We should remove it from the general docs as well then:
https://atlassian.github.io/react-sweet-state/#/advanced/actions?id=async-actions

2nd example

[the remark this answers is not in the document]
Comment (albertogasparin) [reply docs/testing/actions.md]: Oh yes, did I wrote that? 🤦🏼‍♂️

[the remark this answers is not in the document]
Comment (albertogasparin) [reply docs/testing/actions.md]: Oh, I see what you mean. Well, we can explicitly say in the docs that, and still have support for contrived use cases

**Pull request merged**